### PR TITLE
[stable/grafana] fix issue 15134

### LINKED ISSUE #15083: [stable/grafana] unable to import dashboard : download_dashboards.sh failed
**Describe the bug**
I want to upload grafana dashboard  by the id but I'm behind a proxy
even with the env http_proxy,https_proxy,no_proxy set the pod doesn't start
the 

**Version of Helm and Kubernetes**:
K8S 1.14.2

**Which chart**:
Stable/grafana
```
      rbac:
        pspEnabled: true
        create: true
      serviceAccount:
        name: tiller
        create: true

      deploymentStrategy: RollingUpdate
      image:
        repository: grafana/grafana
        tag: 5.2.2
        pullPolicy: IfNotPresent
      securityContext:
        runAsUser: 472
        fsGroup: 472

      downloadDashboardsImage:
        repository: appropriate/curl
        tag: latest
        pullPolicy: IfNotPresent

      plugins:
      persistence:
        enabled: false

      adminUser: admin
      adminPassword:XXXX
      grafana.ini:
        server:
          root_url: https://ingress-mydomain/grafana/
      dashboards:
        default:
          kubernetes-cluster:
            gnetId: 7249
            datasource: Prometheus
          nginx-ingress:
            gnetId: 9614
            datasource: Prometheus
          kubernetes-cluster-monitoring-prometheus:
            gnetId: 8588
            datasource: Prometheus
      extraEmptyDirMounts:
      - name: dashboards
        mountPath: /var/lib/grafana/dashboards/default
      sidecar:
        imagePullPolicy: IfNotPresent
        datasources:
          enabled: true
          label: grafana_datasource
        dashboards:
          enabled: true
          label: grafana_dashboard
      env:
        https_proxy: ((https_proxy))
        http_proxy: ((http_proxy))
        no_proxy: ((no_proxy))
        HTTPS_PROXY: ((https_proxy))
        HTTP_PROXY: ((http_proxy))
        NO_PROXY: ((no_proxy))


```
**What happened**:

The grafana deployment fail
image

the download-dashboard container is not able to download dashboard :

```
kubectl logs grafana-5fc4b5946f-thm2k -c download-dashboards -n monitoring
/etc/grafana/download_dashboards.sh: line 8: can't create /var/lib/grafana/dashboards/default/kubernetes-cluster.json: nonexistent directory
```

**What you expected to happen**:
I espect that dashboards can be uploaded even if my K8S is located behind a proxy

### LINKED ISSUE #15134: [stable/grafana] syntax error in deployment.yaml in grafana 3.5.6
**Describe the bug**
Today, a new helm chart for grafana was release.  This was in it:

https://github.com/helm/charts/blob/9e0d2f23ced8ee73f51a09420b45b097bc89ab04/stable/grafana/templates/deployment.yaml#L78

It caused the YAML parser much chagrin.  Those quotes around the key name are no bueno.  The next line "args": also has the same issue.  Seems like it may have been a bug in converting from JSON to YAML?

**Version of Helm and Kubernetes**:
v2.14.1 and EKS v1.12.6

**Which chart**:  grafana

**What happened**:  YAML parser fail, doesn't install, rolled back to 3.5.4 I believe from 3.5.6

**What you expected to happen**: Things installed

**How to reproduce it** (as minimally and precisely as possible):  Just try to do a `helm install grafana/stable -n grafana --namespace grafana` and watch it go.  Make sure do do the repo update and the delete --purge first.

**Anything else we need to know**:  Not really.

### PR DESCRIPTION
**Pull request by obeyler**
#### What this PR does / why we need it:
fix issue introduce by last PR  https://github.com/helm/charts/pull/15109

#### Which issue this PR fixes
  - fixes #15134
  - fixes #15083

#### Special notes for your reviewer:
sorry for my last PR I don't understand why the "" have been added

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by benmyb** [comment]
But I think there are still commas (this things `,`). Yaml based file should not have `,` to separeate.

File: stable/grafana/templates/deployment.yaml [hunk 1]
```diff
@@ -75,8 +75,8 @@ spec:
         - name: download-dashboards
           image: "{{ .Values.downloadDashboardsImage.repository }}:{{ .Values.downloadDashboardsImage.tag }}"
           imagePullPolicy: {{ .Values.downloadDashboardsImage.pullPolicy }}
-          "command": ["/bin/sh"],
-          "args": [ "-c", "mkdir -p /var/lib/grafana/dashboards/default", "/etc/grafana/download_dashboards.sh" ],
+          command: ["/bin/sh"],
```
Comment (benmyb) [inline stable/grafana/templates/deployment.yaml]: No comma at the end
```suggestion
          command: ["/bin/sh"]
```

File: stable/grafana/templates/deployment.yaml [hunk 2]
```diff
@@ -75,8 +75,8 @@ spec:
...
+          args: [ "-c", "mkdir -p /var/lib/grafana/dashboards/default", "/etc/grafana/download_dashboards.sh" ],
```
Comment (benmyb) [inline stable/grafana/templates/deployment.yaml]: No comma at the end
```suggestion
          args: [ "-c", "mkdir -p /var/lib/grafana/dashboards/default", "/etc/grafana/download_dashboards.sh" ]
```

Comment (obeyler) [reply stable/grafana/templates/deployment.yaml]: thanks @benmyb

File: stable/grafana/templates/deployment.yaml (hunk 1 above)
Comment (obeyler) [reply stable/grafana/templates/deployment.yaml]: thanks !

**Review by zanhsieh — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Comment by qrilka** [comment]
@obeyler were you able to download dashboard this way? I couldn't and it looks to me that this PR isn't completely correct:
`sh -c "mkdir -p /var/lib/grafana/dashboards/default" "/etc/grafana/download_dashboards.sh"` will not invoke `/etc/grafana/download_dashboards.sh` and according to `man sh`:
```
       −c        Read  commands  from  the  command_string operand. Set the value of special parameter 0 (see Section 2.5.2, Special
                 Parameters) from the value of the command_name operand and the  positional  parameters  ($1,  $2,  and  so  on)  in
                 sequence from the remaining argument operands. No commands shall be read from the standard input.
```
it will just set `$0` to `/etc/grafana/download_dashboards.sh` when running `mkdir`. Proper way would be somethig like `sh -c "mkdir -p /var/lib/grafana/dashboards/default" && "/etc/grafana/download_dashboards.sh"`

**Comment by Kieran-Brown** [comment]
@qrilka I can't get it to work either. Latest version I can get to work is 3.5.5 which is prior to the download-dashboards init container change.

Agree with @qrilka. This should be:

```
command: ["/bin/sh"]
args: [ "-c", "mkdir -p /var/lib/grafana/dashboards/default && /etc/grafana/download_dashboards.sh" ]
```

**Comment by obeyler** [comment]
@qrilka @Kieran-Brown  Thanks for your remark I fix it with a new PR

**Pull request merged**